storage/archive/ha_archive.cc

```diff
@@ -1228,7 +1228,7 @@ int ha_archive::rnd_init(bool scan)
       DBUG_RETURN(HA_ERR_CRASHED_ON_USAGE);
 
   if (init_archive_reader())
-      DBUG_RETURN(errno);
+      DBUG_RETURN(HA_ERR_CRASHED_ON_USAGE);
```

Comment:
Please don't change errno to HA_ERR_CRASHED_ON_USAGE.
This will hide the original error !